### PR TITLE
API-954: Add integration tests on each page

### PR DESCRIPTION
**Pull request by pchasle**
**Description (for Contributor and Core Developer)**

Integration tests for each page of Connectivity/Connection/Settings

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/front/src/common/components/__mocks__/Select2.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React, {ChangeEvent} from 'react';
+import {Select2 as BaseSelect2, Select2Configuration} from '../Select2';
+
+export {Select2Configuration};
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/src/common/components/__mocks__/Select2.tsx]: You need to export it there ?

File: src/Akeneo/Connectivity/Connection/front/src/settings/components/ConnectionCreateForm.tsx [hunk 2]
```diff
@@ -77,6 +78,7 @@ const useFormValidation = (
 };
 
 export const ConnectionCreateForm = () => {
+    const history = useHistory();
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/src/settings/components/ConnectionCreateForm.tsx]: Why did you move it from `use-create-connection.ts` to there ?

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/components/__snapshots__/Connection.spec.tsx.snap [hunk 3]
```diff
@@ -14,7 +14,7 @@ exports[`Connection should render 1`] = `
       <img
         alt="Google Shopping"
         className="sc-bxivhb fHwuCK AknImage-display"
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/tests/unit/settings/components/__snapshots__/Connection.spec.tsx.snap]: The className from styled-components are consistent ? I didn't know. It's a hash ? It's not an issue for small css fixes ? or it's on purpose ?

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            <Router history={history}>
+                <Route path='/connections/create'>
+                    <ConnectionsProvider>
+                        <CreateConnection />
+                    </ConnectionsProvider>
+                </Route>
+            </Router>
+        );
+
+        const labelInput = getByLabelText(/^akeneo_connectivity\.connection\.connection\.label/) as HTMLInputElement;
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx]: Why do you need to pass a regex ?

File: src/Akeneo/Connectivity/Connection/front/tests/utils/test-utils.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import {ThemeProvider} from 'styled-components';
+import {theme} from '../../src/common/theme';
+
+const DefaultProviders = ({children}: PropsWithChildren<{}>) => {
+    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
+};
+
+const customRender = (ui: ReactElement) => render(ui, {wrapper: DefaultProviders});
+
+export * from '@testing-library/react';
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/tests/utils/test-utils.tsx]: You don't want to expose react testing library directly ?

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        const codeInput = getByLabelText(/^akeneo_connectivity\.connection\.connection\.code/) as HTMLInputElement;
+        const flowTypeSelect = getByText('akeneo_connectivity.connection.flow_type.data_source')
+            .parentElement as HTMLSelectElement;
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx]: You cannot access the `select` directly ?

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            userEvent.selectOptions(flowTypeSelect, 'data_destination');
+            userEvent.click(saveButton);
+        });
+
+        expect(labelInput.value).toBe('Magento');
+        expect(codeInput.value).toBe('magento');
+        expect(flowTypeSelect.value).toBe('data_destination');
+
+        expect(fetchMock).toBeCalled();
+        expect(fetchMock.mock.calls[0][0]).toEqual('akeneo_connectivity_connection_rest_create');
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx]: Why don't you use `toHaveBeenCalledWith` ?

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/DeleteConnection.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                        <DeleteConnection />
+                    </ConnectionsProvider>
+                </Route>
+            </Router>
+        );
+
+        const deleteButton = getByText('pim_common.delete');
+
+        await act(async () => {
+            userEvent.click(deleteButton);
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/DeleteConnection.spec.tsx]: Why do you prefer `userEvent from '@testing-library/user-event'` over `import {fireEvent} from '@testing-library/react'` ?
In our tests, I don't need to await act around fireEvent but we don't have async calls behind.

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/EditConnection.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+jest.mock('../../../../src/common/components/Select2');
+
+describe('testing EditConnection page', () => {
+    beforeEach(() => {
+        fetchMock.resetMocks();
+    });
+
+    it('creates a connection', async () => {
+        fetchMock.mockResponseOnce(
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/EditConnection.spec.tsx]: It's not risky to hardcode in which order the fetch calls will be passed ? Do you need to ensure they will keep this order ?

File: src/Akeneo/Connectivity/Connection/front/src/common/components/__mocks__/Select2.tsx (hunk 1 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/src/common/components/__mocks__/Select2.tsx]: To be safe and have the same interface on my mocked module. I don't think it's necessary though.

File: src/Akeneo/Connectivity/Connection/front/src/settings/components/ConnectionCreateForm.tsx (hunk 2 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/src/settings/components/ConnectionCreateForm.tsx]: I had the error "update of the state of a null component" (or something like that), the order of execution was not right. Redirection triggered before state update.
+ it's better to have the hook doing the API call and not handling any redirection, has it's more a responsibility of the component calling it.

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/components/__snapshots__/Connection.spec.tsx.snap (hunk 3 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/tests/unit/settings/components/__snapshots__/Connection.spec.tsx.snap]: Yes, it's a hash and will be different after each change, not much of an issue, it's more explicit to see that css has been modified.

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx (hunk 4 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx]: Because the label has a 'required' text appended to it in the text node.

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx (hunk 6 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx]: I tried to setup a `for=` on the label of the select (and `id=` on the select) but I needed to refactor the Select2 mock (and change his API)... not worth it, in my opinion.

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx (hunk 7 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/CreateConnection.spec.tsx]: I want to use `toMatchObject` on the second argument to not have to compare the whole object (not the headers, etc...)

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/DeleteConnection.spec.tsx (hunk 8 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/DeleteConnection.spec.tsx]: It was suggested in one of the documentation. `userEvent` is doing all the fireEvent logic and is much more concise, no need to create event manually, and it's way easier for typing whole words/sentences.

File: src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/EditConnection.spec.tsx (hunk 9 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/tests/unit/settings/pages/EditConnection.spec.tsx]: Probably not ideal but I don't know any other way.

File: src/Akeneo/Connectivity/Connection/front/tests/utils/test-utils.tsx (hunk 5 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/tests/utils/test-utils.tsx]: This whole snippet is something I took from testing-library, I don't mind either one here.
